cubano-concordion/src/main/java/org/concordion/cubano/framework/ConcordionBase.java

```diff
@@ -37,8 +37,8 @@
     private static List<Browser> allBrowsers = new ArrayList<Browser>();
     private static ThreadLocal<Map<String, Browser>> threadBrowsers = ThreadLocal.withInitial(HashMap::new);
     private static ThreadLocal<Integer> testCount = ThreadLocal.withInitial(() -> 0);
+    private static ThreadLocal<Boolean> browserTestRunCounted = ThreadLocal.withInitial(() -> false);
```

Comment:
It feels like this class is doing too much now and violating the Single Responsibility Principle. I'd suggest moving a lot of the browser handling code to a class such as ActiveBrowsers, BrowserList or BrowserManager (though I hate having classes called *Manager!).